File: src/main/java/com/zowiac/commons/DataValues.java
package com.zowiac.commons;

import java.util.HashMap;
import java.util.Map;

public class DataValues {

    private static Map<String, String> populationMap, forecastPopulationMap;

    public static Map<String, String> getPopulationMap() {
        if (populationMap == null) {
            populationMap = new HashMap<>();
            populationMap.put("30", "sehr häufig");
            populationMap.put("25", "häufig");
            populationMap.put("20", "mäßig häufig");
            populationMap.put("15", "selten");
            populationMap.put("10", "sehr selten");
            populationMap.put("5", "extrem selten");
            populationMap.put("0", "ausgestorben oder verschollen");
        }
        return populationMap;
    }

    public static Map<String, String> getForecastPopulationMap() {
        if (forecastPopulationMap == null) {
            forecastPopulationMap = new HashMap<>();
            forecastPopulationMap.put("30", "Rückgang, im Ausmaß unbekannt");
            forecastPopulationMap.put("25", "sehr starker Rückgang");
            forecastPopulationMap.put("20", "starker Rückgang");
            forecastPopulationMap.put("15", "mäßiger Rückgang");
            forecastPopulationMap.put("10", "stabil");
            forecastPopulationMap.put("5", "deutliche Zunahme");
            forecastPopulationMap.put("0", "Daten ungenügend");
        }
        return forecastPopulationMap;
    }


}

File: src/main/webapp/app/app/commons/DataValues.js
Ext.define('zowiac.commons.DataValues', {
    statics: {
        states: [
            ['BU', 'bundesweit'],
            ['BW', 'Baden-Württemberg'],
            ['BY', 'Bayern'],
            ['BE', 'Berlin'],
            ['BB', 'Brandenburg'],
            ['HB', 'Bremen'],
            ['HH', 'Hamburg'],
            ['HE', 'Hessen'],
            ['MV', 'Mecklenburg-Vorpommern'],
            ['NI', 'Niedersachsen'],
            ['NW', 'Nordrhein-Westfalen'],
            ['RP', 'Rheinland-Pfalz'],
            ['SL', 'Saarland'],
            ['SN', 'Sachsen'],
            ['ST', 'Sachsen-Anhalt'],
            ['SH', 'Schleswig-Holstein'],
            ['TH', 'Thüringen']
        ],

        population: [
            ['30', 'sehr häufig'],
            ['25', 'häufig'],
            ['20', 'mäßig häufig'],
            ['15', 'selten'],
            ['10', 'sehr selten'],
            ['5', 'extrem selten'],
            ['0', 'ausgestorben oder verschollen']
        ],

        forecastPopulation: [
            ['30', 'Rückgang, im Ausmaß unbekannt'],
            ['25', 'sehr starker Rückgang'],
            ['20', 'starker Rückgang'],
            ['15', 'mäßiger Rückgang'],
            ['10', 'stabil'],
            ['5', 'deutliche Zunahme'],
            ['0', 'Daten ungenügend']
        ],
        reportType: [
            ['J', 'ZOWIAC'],
            ['H', 'Jagd'],
            ['O', 'Beobachter']],

        yesNo: [['J', 'Ja'], ['N', 'Nein']],

        getValueByKey(list, key) {
            try {
                for (let i = 0; i < list.length; i++)
                    if (list[i][0] === key)
                        return list[i][1];
            } catch (e) {
            }
            return '';

        }
    }
});
